server/storage/mvcc/watchable_store.go

```diff
@@ -482,6 +483,26 @@ func (s *watchableStore) progress(w *watcher) {
 	}
 }
 
+func (s *watchableStore) progress_all(watchers map[WatchID]*watcher) bool {
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
I think we only need to require read lock here, in the same way as the existing `(*watchableStore) progress`

```suggestion
	s.mu.RLock()
	defer s.mu.RUnlock()
```